Generation/src/components/HepMCFileReaderTool.h

```diff
@@ -0,0 +1,31 @@
+
+#ifndef GENERATION_HEPMCFILEREADER_H 
+#define GENERATION_HEPMCFILEREADER_H
+
+#include "GaudiAlg/GaudiTool.h"
+#include "Generation/IHepMCFileReaderTool.h"
+
+#include "HepMC/GenEvent.h"
+#include "HepMC/IO_GenEvent.h"
+
+class HepMCFileReader: public GaudiTool, virtual public IHepMCFileReaderTool {
+public:
+  HepMCFileReader( const std::string& type, const std::string& name,
+    const IInterface* parent) ;
+  virtual ~HepMCFileReader(); ///< Destructor
+  virtual StatusCode initialize();
+  virtual StatusCode finalize();
+
+  /// Wrapper for HepMC's fill_next_event() --
+  /// ownership of the event is given to the data service (in algorithm HepMCReader)
```

Comment:
Maybe rephrase this, as somebody might want to use the tool in another algorithm.. Something like "the user is responsible for the deletion of the event"